hibernate-core/src/main/java/org/hibernate/type/descriptor/java/AbstractArrayJavaType.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.type.descriptor.java;
+
+import java.sql.Types;
+
+import org.hibernate.dialect.Dialect;
+import org.hibernate.engine.jdbc.spi.JdbcServices;
+import org.hibernate.tool.schema.extract.spi.ColumnTypeInformation;
+import org.hibernate.type.BasicArrayType;
+import org.hibernate.type.BasicPluralType;
+import org.hibernate.type.BasicType;
+import org.hibernate.type.descriptor.jdbc.ArrayJdbcType;
+import org.hibernate.type.descriptor.jdbc.JdbcType;
+import org.hibernate.type.descriptor.jdbc.JdbcTypeIndicators;
+import org.hibernate.type.spi.TypeConfiguration;
+
+public abstract class AbstractArrayJavaType<T, E> extends AbstractClassJavaType<T>
```

Comment:
I'd also rename this one to reflect whatever naming you choose with the "basic plurality".  Just based on name, I have no idea that this is limited to these basic pluralities compared to plural associations.

Reply:
Not fond of he `AbstractClassJavaType` name either in retrospect.  But probably not worth changing now